### ISSUE TITLE
Cannot set service name in zipkin with properties with Spring Boot 3.1

### ISSUE THREAD

**Issue by chillb0nes** [spring-projects/spring-boot#36301]
In spring-cloud-sleuth there was a property `spring.zipkin.service.name` which could be used to set span sender name that will appear in Zipkin. After sleuth decommission and upgrading to Spring Boot 3.1, the actuator autoconfiguration module always sets it to `spring.application.name`.

https://github.com/spring-projects/spring-boot/blob/3.1.x/spring-boot-project/spring-boot-actuator-autoconfigure/src/main/java/org/springframework/boot/actuate/autoconfigure/tracing/BraveAutoConfiguration.java#L121

I solved this issue by creating a custom `TracingCustomizer` bean in my configuration, but configuration via properties looks more convenient. What do you think?

**Comment by wilkinsona** [comment]
I assume that you can't set `spring.application.name` for some reason. Can you please explain why that's not an option? Without such an explanation it will be hard to justify introducing a second property.

**Comment by wilkinsona** [comment]
Thanks.

In addition to `BraveAutoConfiguration`, there's also similar usage with Otel:

https://github.com/spring-projects/spring-boot/blob/7e5bfc4b2e88eb4a841df3108ee4344d2e486817/spring-boot-project/spring-boot-actuator-autoconfigure/src/main/java/org/springframework/boot/actuate/autoconfigure/tracing/OpenTelemetryAutoConfiguration.java#L105-L108

`spring.application.name` is also used with `Wavefront` but only as a fallback if `management.wavefront.application.service-name` hasn't been set:

https://github.com/spring-projects/spring-boot/blob/7e5bfc4b2e88eb4a841df3108ee4344d2e486817/spring-boot-project/spring-boot-actuator-autoconfigure/src/main/java/org/springframework/boot/actuate/autoconfigure/wavefront/WavefrontAutoConfiguration.java#L49-L54

Lastly, on the metrics side of things, there's also some usage with Prometheus:

https://github.com/spring-projects/spring-boot/blob/7e5bfc4b2e88eb4a841df3108ee4344d2e486817/spring-boot-project/spring-boot-actuator-autoconfigure/src/main/java/org/springframework/boot/actuate/autoconfigure/metrics/export/prometheus/PrometheusMetricsExportAutoConfiguration.java#L149-L153

Wavefront feels like an odd one out here and I think it would be good to remove that inconsistency. We'll need to decide how we want to do that. Should it be a single property that we start using with Wavefront instead of `management.wavefront.application.service-name` or should there be several new properties, each only used in one place.

**Comment by chillb0nes** [comment]
Hi @jonatan-ivanov,

That's a nice option too, thanks. The application name in our code is most often used together with the namespace, but they are provided as separate fields, parameters, etc. Providing the namespace both in the service name and in a separate field just because it is more handy to search traces in Zipkin seems superfluous.

**Comment by jonatan-ivanov** [comment]
>Providing the namespace both in the service name and in a separate field just because it is more handy to search traces in Zipkin seems superfluous.

I might not be perfectly clear on this, I would _not_ encode the namespace in the app name, I would only add a tag with the namespace and leave the app name being only the name of the app. To me encoding the namespace into the app name seems quite hacky, confusing and it does not reflect reality. Adding the namespace separately as a tag does reflect reality (every app has a name and a namespace). That way it is easier to search for the app name or the namespace or both together. Also, you don't need to know the specific format, e.g.: if one sees `abc:xyz`, which one is the namespace/app name?

This is true both for tracing and metrics, please take a look at [Micrometer docs](https://micrometer.io/docs/concepts#_supported_monitoring_systems) and check the section about dimensional vs. hierarchical backends (it's about metrics but please bear with me). Dimensional systems add data as separate key-value pairs (~like a hashmap) while hierarchical systems are trying to encode everything into the name (~like a single string). Zipkin has key-values (tags), additional metadata should be added as such and not appended into a single string.

**Comment by chillb0nes** [comment]
I got it, I answered this question:

> I would be curious what are you using the app name for in other parts of the codebase that prevents you adding whatever app name you want.

Adding namespace as a tag is a nice option. I guess it just happened historically that the namespace was encoded in the service name for Zipkin in our code because spring-cloud-sleuth provides a property for that and it came in handy.

I feel that property-based customization is easier, and it also could make configurability of OTel and Brave consistent with Wavefront. But too many properties is not good too. If using different names for the spring application and service in the monitoring system is not a common pattern maybe, it might be better to leave it as it is.
